### PR TITLE
fix: regression caused by #6376

### PR DESCRIPTION
**Pull request by hubertzub-db**
Signed-off-by: Hubert Zub <hubert.zub@databricks.com>

## What changes are proposed in this pull request?

Fixes regression caused by https://github.com/mlflow/mlflow/pull/6376
Propagates target step name by an environment variable now.

## How is this patch tested?
- Manual tests
- Fixed existing unit test

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [x] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/pipelines/test_train_step.py [hunk 1]
```diff
@@ -134,7 +134,11 @@ def test_train_steps_autologs(tmp_pipeline_root_path):
 
 def test_train_steps_with_correct_tags(tmp_pipeline_root_path):
     with mock.patch.dict(
-        os.environ, {_MLFLOW_PIPELINES_EXECUTION_DIRECTORY_ENV_VAR: str(tmp_pipeline_root_path)}
+        os.environ,
+        {
+            _MLFLOW_PIPELINES_EXECUTION_DIRECTORY_ENV_VAR: str(tmp_pipeline_root_path),
+            "target_step_name": "train",
```
Comment (liangz1) [inline tests/pipelines/test_train_step.py:140]: What about defining this env var key as `_MLFLOW_PIPELINES_TARGET_STEP_NAME_ENV_VAR` (following the pattern above)?

Comment (dbczumar) [reply tests/pipelines/test_train_step.py:140]: +1

**Review by dbczumar — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/6689/files#r961919688 is addressed. Thanks @hubertzub-db !

File: tests/pipelines/test_train_step.py (hunk 1 above)
Comment (hubertzub-db) [reply tests/pipelines/test_train_step.py]: good point, done!

File: mlflow/pipelines/utils/execution.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 
 _MLFLOW_PIPELINES_EXECUTION_DIRECTORY_ENV_VAR = "MLFLOW_PIPELINES_EXECUTION_DIRECTORY"
+_MLFLOW_PIPELINES_EXECUTION_TARGET_STEP_NAME_ENV_VAR = "target_step_name"
```
Comment (liangz1) [inline mlflow/pipelines/utils/execution.py:13]:
```suggestion
_MLFLOW_PIPELINES_EXECUTION_TARGET_STEP_NAME_ENV_VAR = "MLFLOW_PIPELINES_EXECUTION_TARGET_STEP_NAME"
```

**Review by liangz1 — APPROVED** [review]
LGTM pending one minor comment. Thanks!

**Pull request merged**